cachito/web/status.py

```diff
@@ -0,0 +1,142 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import requests
+from sqlalchemy.exc import SQLAlchemyError
+
+from cachito.errors import CachitoError
+from cachito.web import db
+from cachito.workers.config import get_worker_config
+from cachito.workers.tasks.celery import app
+
+
+no_retry_session = requests.Session()
+
+
+def _service_ok(url):
+    """
+    Check if a service is reachable and working correctly.
+
+    The status of the service is determined from the status code of the response.
+
+    :param str url: url to fetch
+    :return: tuple (ok: bool, reason: str or None)
+    """
+    try:
+        resp = no_retry_session.get(url)
+        resp.raise_for_status()
+    except requests.RequestException as e:
+        return False, str(e)
+
+    return True, None
+
+
+def nexus_ok(nexus_url):
+    """
+    Check if Nexus is OK.
+
+    Uses the /status endpoint, specifically /status/writable.
+
+    :param str nexus_url: url of the Nexus host
+    :return: tuple (ok: bool, reason: str or None)
+    """
+    nexus_status_url = f"{nexus_url.rstrip('/')}/service/rest/v1/status/writable"
+    return _service_ok(nexus_status_url)
+
+
+def athens_ok(athens_url):
+    """
+    Check if Athens is OK.
+
+    Simply checks if the root url is reachable (Athens has no status endpoint).
```

Comment:
Note for future selves: This may be useful in the future https://github.com/gomods/athens/pull/1658